### PR TITLE
Move get_http_response to asgi class to make it easier to override it

### PR DESCRIPTION
**Pull request by patrick91**
## Description

I need to override graphiql for an internal service but without this PR the changes is way too invasive.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/),so make sure to pick the appropriate type. If in doubt feel free to ask :)

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: patch
```
Comment (BryceBeagle) [inline RELEASE.md:1]: Minor? Do we expect anyone to have been using this function?

Comment (patrick91) [reply RELEASE.md:1]: no, I don't think anyone was using it :)

Comment (patrick91) [reply RELEASE.md:1]: I'm going to do minor just in case :D

Comment (BryceBeagle) [reply RELEASE.md]: Might want to make the change clear in the RELEASE.md too? (i.e. "moved x from y to z")

Comment (patrick91) [reply RELEASE.md]: will do :)

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Release type: minor
+
+This releases updates the ASGI class to make it easier to override `get_http_response`.
```
Comment (patrick91) [inline RELEASE.md:3]:
```suggestion
This releases updates the ASGI class to make it easier to override `get_http_response`.

`get_http_response` has been now removed from strawberry.asgi.http and been moved to be
a method on the ASGI class. 

A new `get_graphiql_response` method has been added to make it easier to provide a different GraphiQL interface.
```

Comment (BryceBeagle) [reply RELEASE.md:3]: Looks good to me 👍

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**